src/ape/contracts/base.py

```
@@ -284,43 +284,61 @@ def __getattr__(self, attr_name: str) -> Any:
             any: The return value from the contract call, or a transaction receipt.
         """
 
-        handlers = {
-            "events": ContractEvent,
-            "calls": ContractCallHandler,
-            "transactions": ContractTransactionHandler,
-        }
+        def name_matches(abi):
+            return abi.name == attr_name
+
+        selected_view_methods = list(filter(name_matches, self._contract_type.view_methods))
+        has_matching_view_methods = len(selected_view_methods) > 0
+
+        selected_mutable_methods = list(filter(name_matches, self._contract_type.mutable_methods))
+        has_matching_mutable_methods = len(selected_mutable_methods) > 0
+
+        selected_events = list(filter(name_matches, self._contract_type.events))
```

should we raise an error if for some reason we match multiple sets here?